compiler/rustc_codegen_ssa/src/back/linker.rs

```diff
@@ -178,23 +181,53 @@ fn verbatim_args<L: Linker + ?Sized>(
     }
     l
 }
+/// Add underlying linker arguments to C compiler command, by wrapping them in
+/// `-Wl` or `-Xlinker`.
+fn convert_link_args_to_cc_args(
+    cmd: &mut Command,
+    args: impl IntoIterator<Item: AsRef<OsStr>, IntoIter: ExactSizeIterator>,
+) {
+    let args = args.into_iter();
+    if args.len() == 0 {
+        return;
+    }
+
+    let mut combined_arg = OsString::from("-Wl");
+    for arg in args {
+        // If the argument itself contains a comma, we need to emit it
+        // as a `-Xlinker` argument instead.
```

Comment:
Nit: the "instead" looks strange with the current order of if-else branches, maybe put the good path first?

Reply:
I've tried to write something clearer, let me know what you think?